scripts/getsamplesummary.py

```diff
@@ -1,90 +1 @@
-#!/usr/bin/python
-#
-import sys
-import datetime
-import time
-import glob
-import re
-import socket
-import os
-import select
-from access import db, lims
-
-outputdir = '/mnt/hds/proj/bioinfo/tmp/MIPP/exomes/'
-
-fc = "flowcell"
-if len(sys.argv) > 0:
-  try:
-    sys.argv[1]
-  except NameError:
-    sys.exit("Usage: " + sys.argv[0] + " <flowcell name>")
-  else:
-    fc = sys.argv[1]
-else:
-  sys.exit("Usage: " + sys.argv[0] + " <flowcell name>")
-
-params = db.readconfig("non")
-
-fc_samples = {}
-def getsamplesfromflowcell(pars, flwc):
-  samples = glob.glob(pars['DEMUXDIR'] + "*" + flwc + "*/Unaligned/Project_*/Sample_*")
-  for sampl in samples:
-    sample = sampl.split("/")[len(sampl.split("/"))-1].split("_")[1]
-    fc_samples[sample] = ''
-  return fc_samples
-
-def getsampleinfofromname(pars, sample):
-  query = (" SELECT sample.sample_id AS id, samplename, flowcellname AS fc, " + 
-           " lane, ROUND(readcounts/2000000,2) AS M_reads, " +
-           " ROUND(q30_bases_pct,2) AS q30, ROUND(mean_quality_score,2) AS score " + 
-           " FROM sample, unaligned, flowcell " + 
-           " WHERE sample.sample_id = unaligned.sample_id AND unaligned.flowcell_id = flowcell.flowcell_id " +
-           " AND (samplename LIKE '" + sample + "_%' OR samplename = '" + sample + "')")
-  with db.create_tunnel(pars['TUNNELCMD']):
-    with db.dbconnect(pars['CLINICALDBHOST'], pars['CLINICALDBPORT'], pars['STATSDB'], 
-                   pars['CLINICALDBUSER'], pars['CLINICALDBPASSWD']) as dbc:
-      replies = dbc.generalquery( query )
-  return replies
-
-def makelinks(samplename, lanedict):
-  if os.path.exists(outputdir + samplename):
-    print outputdir + samplename + ' exists, has data already been exported?'
-    return
-  else:
-    os.makedirs(outputdir + samplename)
-    for entry in lanedict:
-      fclane = lanedict[entry].split("_")
-      print fclane
-      fastqfiles = glob.glob(params['DEMUXDIR'] + "*" + fclane[0] + "*/Unaligned/Project_*/Sample_*" + 
-                            samplename + "_*/*L00" + fclane[2] + "*gz")
-      for fastqfile in fastqfiles:
-        nameparts = fastqfile.split("/")[len(fastqfile.split("/"))-1].split("_")
-        date_fc = fastqfile.split("/")[6].split("_")[0] + "_" + fastqfile.split("/")[6][-9:]
-        newname = (nameparts[3][-1:] + "_" + date_fc + "_" + samplename + "_" + nameparts[2] +
-                   "_" + nameparts[4][-1:] + ".fastq.gz")
-        print fastqfile
```

Comment:
A more flexible logging solution than naked print statements would be using Python's logging module. It's very easy to use.

```python
import logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

logger.info('Start reading database')
```

Read more here: http://victorlin.me/posts/2012/08/26/good-logging-practice-in-python